Library/Homebrew/extend/os/linux/tap.rb

```diff
@@ -1,9 +1,7 @@
 class CoreTap < Tap
-  def default_remote
-    if ENV["HOMEBREW_FORCE_HOMEBREW_ON_LINUX"]
-      "https://github.com/Homebrew/homebrew-core".freeze
-    else
-      "https://github.com/Linuxbrew/homebrew-core".freeze
-    end
+  # @private
+  def initialize
+    super "Homebrew", "core"
+    @full_name = "Linuxbrew/homebrew-core" unless ENV["HOMEBREW_FORCE_HOMEBREW_ON_LINUX"]
```

Comment:
I'm not sure this is properly overriding e.g.the path etc.

Reply:
The aim here is to override only `full_name` and `default_remote` on Linux. The other variables should be identical on Linux as on macOS. In particular, `CoreTap.instance.path` is `$HOMEBREW_PREFIX/Homebrew/Library/Taps/homebrew/homebrew-core`.
```console
$ brew irb <<<'CoreTap.instance.path'
…
<Pathname:/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core>
```